demo/grid-columns-demos.html

```diff
@@ -6,6 +6,68 @@
+      The column width will be calculated and set only once based on the contents of the initially rendered cells.
+      You can manually trigger the auto sizing behavior again by calling <code>grid.recalculateColumnWidths()</code>.
+    </p>
+    <p>
+      By default the column might still expand if there is more space available, unless you also set <code>flex-grow="0"</code>.
+    </p>
+    <p>
+      <b>Note:</b> If all the columns have <code>flex-grow="0"</code> there might be empty space after the last column
+      if the grid is wider than what is needed to fit all the columns. You probably want to leave at least one column
+      without <code>flex-grow="0"</code> so that column will take all the remaining space that is left.
+    </p>
+    <p>
+      <b>Note:</b> When using column renderer functions, those should be set before setting the grid <code>items</code> or
+      <code>dataProvider</code>. Otherwise the initial column widths may be calculated before the renderers have been run and
+      the widths of the renderer columns might be wrong initially.
+    </p>
+    <vaadin-demo-snippet id='grid-columns-demos-auto-width'>
+      <template preserve-content>
+        <vaadin-button>Recalculate column widths</vaadin-button>
+        <vaadin-grid theme="compact column-borders" aria-label="Automatic Column Width Example">
+          <vaadin-grid-column width="50px" flex-grow="0" text-align="end" header="#"></vaadin-grid-column>
+          <vaadin-grid-column auto-width flex-grow="0" text-align="center"></vaadin-grid-column>
+          <vaadin-grid-column auto-width path="name.first"></vaadin-grid-column>
+          <vaadin-grid-column auto-width path="name.last"></vaadin-grid-column>
+          <vaadin-grid-column auto-width path="location.city"></vaadin-grid-column>
+          <vaadin-grid-column auto-width path="location.state"></vaadin-grid-column>
+          <vaadin-grid-column auto-width path="location.street"></vaadin-grid-column>
+        </vaadin-grid>
+
+        <script>
+          window.addDemoReadyListener('#grid-columns-demos-auto-width', function(document) {
+            const grid = document.querySelector('vaadin-grid');
+
+            const columns = document.querySelectorAll('vaadin-grid-column');
+            columns[0].renderer = function(root, column, rowData) {
+              root.textContent = rowData.index;
+            };
+            columns[1].renderer = function(root, column, rowData) {
+              if (!root.firstElementChild) {
+                root.innerHTML = '<iron-image width="25" height="25" sizing="cover"></iron-image>';
+              }
+              root.firstElementChild.alt = rowData.item.name.first + ' ' + rowData.item.name.last;
+              root.firstElementChild.src = rowData.item.picture.thumbnail;
+            };
+
+            grid.items = Vaadin.GridDemo.users;
+
+            const button = document.querySelector('vaadin-button');
+            button.addEventListener('click', e => {
```

Comment:
Must use ES5 in the demos to support IE11

Reply:
Ok, will fix and push a commit with [skip ci].

Reply:
Arrow function replaced with normal anonymous function. Demo tested working in IE11 now.